lib/Backends/OpenCL/OpenCLDeviceManager.cpp

```diff
@@ -0,0 +1,197 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#define DEBUG_TYPE "opencl"
+
+// Silence Apple's warning about the deprecation of OpenCL.
+#define CL_SILENCE_DEPRECATION
+
+// Silence warnings about using deprecated OpenCL 1.2 functions.
+#define CL_USE_DEPRECATED_OPENCL_1_2_APIS
+
+#include "OpenCLDeviceManager.h"
+
+#include "llvm/ADT/StringExtras.h"
+#include "llvm/Support/raw_ostream.h"
+
+using namespace glow;
+using namespace glow::runtime;
+
+namespace glow {
+namespace runtime {
+DeviceManager *createOCLDeviceManager(llvm::StringRef name) {
+  return new OpenCLDeviceManager(name);
+}
+} // namespace runtime
+} // namespace glow
+
+cl_mem OpenCLDeviceManager::allocDeviceBuffer(uint64_t size) {
+  const uint64_t alignment = 128;
+  // Always allocate buffers properly aligned to hold values of any type.
+  size = alignedSize(size, alignment);
+  auto buf =
+      clCreateBuffer(context_, CL_MEM_READ_WRITE, size, nullptr, nullptr);
+  GLOW_ASSERT(buf && "Allocation failed!");
+  return buf;
+}
+OpenCLDeviceManager::OpenCLDeviceManager(llvm::StringRef name)
+    : QueueBackedDeviceManager(BackendKind::OpenCL, name) {
+  // For now we have a string, if the first digit is an int use it, otherwise
```

Comment:
We currently have some logic in the OpenCL.cpp, in `OpenCLFunction::OpenCLFunction` to select a proper platform and device. It can be configured via some command-line options. This code seems very similar. Should we add similar options here or may be reuse the existing ones?

Reply:
Once we have executionInfo there will be a migration of functionality from the compiledFunction to the deviceManager, we can move the options over then.